packages/components/datepicker/examples/Datepicker/Custom.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useState, useRef } from 'react';
+import { Calendar } from '@contentful/f36-datepicker';
+import { TextInput } from '@contentful/f36-forms';
+import { format } from 'date-fns';
+import { Popover } from '@contentful/f36-popover';
+import FocusLock from 'react-focus-lock';
+import { css } from 'emotion';
+import tokens from '@contentful/f36-tokens';
+
+export default function CustomExample() {
+  const DATE_FORMAT = 'dd LLL yyyy'; // e.g. 31 Jan 2022
+
+  const [selected, setSelected] = useState();
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+  const [inputValue, setInputValue] = useState(() =>
+    selected ? format(selected, DATE_FORMAT) : '',
+  );
+  const popoverWasClosed = useRef(false);
+
+  const closePopover = () => {
+    popoverWasClosed.current = true;
+    setIsPopoverOpen(false);
+  };
+
+  const handleDaySelect = (date) => {
+    if (date) {
+      setSelected(date);
+      setInputValue(format(date, DATE_FORMAT));
+      closePopover();
+    }
+  };
+
+  const handleInputInteraction = () => {
+    // when popover gets closed focus returns to the input,
+    // we want to prevent popover to be opened in this case
+    if (popoverWasClosed.current) {
+      popoverWasClosed.current = false;
+      return;
+    }
+
+    setIsPopoverOpen(true);
+  };
+
+  return (
+    <Popover isOpen={isPopoverOpen} onClose={closePopover}>
+      <Popover.Trigger>
+        <TextInput
```

Comment:
It's not supposed to work with typing. Since we use the input as a trigger, I don't see any way we can allow users to type date. Do you have ideas? The way it works in the "pikaday" is a pure magic, and not really usable

Reply:
If you close the popover the focus returns to the input but you still can't type, would we want to support that?

Reply:
Adjusted the example to support typing. It made the example a bit more complex and heavier. 
Let me know what you think